src/main/java/seedu/duke/parser/schedule/ParserSchedule.java

```diff
@@ -113,26 +166,19 @@ public static YearMonth parseCalendarCommandForJunit(String input) {
      */
     public static ArrayList<String> parseLectureArgumentsArray(String input)
             throws IncorrectNumberOfArgumentsException, InvalidDateException {
+        checkForDividersAddLectureCommand(input);
         String lectureDetails = input.trim().substring(17);
-
-        String nameAndDate = lectureDetails.substring(lectureDetails.indexOf("m/")).trim();
-        String name = nameAndDate.substring(nameAndDate.indexOf("m/") + 2, nameAndDate.indexOf("s/")).trim();
-        String dayAndLimits = nameAndDate.substring(nameAndDate.indexOf("s/")).trim();
-        String fromDate = dayAndLimits.substring(dayAndLimits.indexOf("s/") + 2, dayAndLimits.indexOf("e/")).trim();
-        String toDate = dayAndLimits.substring(dayAndLimits.indexOf("e/") + 2).trim();
-        if (name.equals("")) {
-            throw new IncorrectNumberOfArgumentsException("Module name not found after m/...");
-        }
-        if (fromDate.equals("")) {
-            throw new IncorrectNumberOfArgumentsException("Module name not found after m/...");
-        } else if (fromDate.length() != 10) {
-            throw new InvalidDateException();
-        }
-        if (toDate.equals("")) {
-            throw new IncorrectNumberOfArgumentsException("Module end date not found after e/...");
-        } else if (toDate.length() != 10) {
-            throw new InvalidDateException();
-        }
+        int moduleIndexFormer = lectureDetails.indexOf(MODULE_DIVIDER);
+        String nameAndDate = lectureDetails.substring(moduleIndexFormer).trim();
+        int moduleIndexLatter = nameAndDate.indexOf(MODULE_DIVIDER);
```

Comment:
The variable names could be more specific. Hard to understand the purpose of moduleIndexLatter and also nameAndDate could be something like nameAndDateOfLecture.